### PR TITLE
Feature/jaccard similarity

### PR DESCRIPTION
**Pull request by Owlthority**
Hereby a pull request for the added jacard similarity feature, it has passed all 7 tests in 'pytest tests.py'

### CODE REVIEW

File: table_evaluator/table_evaluator.py [hunk 1]
```diff
@@ -46,6 +43,7 @@ def __init__(self, real: pd.DataFrame, fake: pd.DataFrame, cat_cols=None, unique
         self.fake = fake.copy()
         self.comparison_metric = getattr(stats, metric)
         self.verbose = verbose
+        self.random_seed = seed
```
Comment (Baukebrenninkmeijer) [inline table_evaluator/table_evaluator.py:46]: I'd add the below mentioned seed setting here.

File: table_evaluator/TEST_jac_sim.ipynb [hunk 2]
```diff
@@ -0,0 +1,68 @@
+{
+ "cells": [
```
Comment (Baukebrenninkmeijer) [inline table_evaluator/TEST_jac_sim.ipynb:2]: Make sure that adding the notebook adds value. Currently, I think it just shows an error, which is not adding that much. So either put some other things here or remove the notebook.

File: table_evaluator/table_evaluator.py [hunk 3]
```diff
@@ -445,7 +461,10 @@ def estimator_evaluation(self, target_col: str, target_type: str = 'class') -> f
             fake_y = self.fake[target_col]
         else:
             raise Exception(f'Target Type must be regr or class')
-
+        
+        # For reproducibilty:
+        np.random.seed(self.random_seed)
```
[the remark this answers is not in the document]
Comment (Owlthority) [reply table_evaluator/table_evaluator.py:466]: I had the same thought when coding, however for reasons that are unclear to me it did not produce similar results. It had to placed specifically here.

File: table_evaluator/TEST_jac_sim.ipynb (hunk 2 above)
Comment (Owlthority) [reply table_evaluator/TEST_jac_sim.ipynb:2]: I will remove this one.

File: table_evaluator/table_evaluator.py (hunk 1 above)
Comment (Owlthority) [reply table_evaluator/table_evaluator.py:46]: It had to be placed specifically before those train_test_split functions in order to produce reproducible results.

File: table_evaluator/table_evaluator.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Baukebrenninkmeijer) [reply table_evaluator/table_evaluator.py:466]: Hmm, strange.

File: table_evaluator/table_evaluator.py [hunk 4]
```diff
@@ -319,9 +318,26 @@ def score_estimators(self):
             f2r = [rmse(self.real_y_test, clf.predict(self.real_x_test)) for clf in self.f_estimators]
             index = [f'real_data_{classifier}' for classifier in self.estimator_names] + \
                     [f'fake_data_{classifier}' for classifier in self.estimator_names]
-            results = pd.DataFrame({'real': r2r + r2f, 'fake': f2r + f2f}, index=index)
+            results = pd.DataFrame({'real': r2r + r2f, 'fake': f2r + f2f}, index=index)      
         else:
             raise Exception(f'self.target_type should be either \'class\' or \'regr\', but is {self.target_type}.')
+
+        rows = []
+        jac_sim = []
```
Comment (Baukebrenninkmeijer) [inline table_evaluator/table_evaluator.py:326]: This can be removed I think. Seems to be from old code.

Comment (Baukebrenninkmeijer) [inline table_evaluator/table_evaluator.py:326]: This line can be removed I think. Seems to be from old code.

File: table_evaluator/table_evaluator.py [hunk 5]
```diff
@@ -474,9 +493,11 @@ def estimator_evaluation(self, target_col: str, target_type: str = 'class') -> f
         for estimator in self.estimators:
             assert hasattr(estimator, 'fit')
             assert hasattr(estimator, 'score')
-
+        
         self.fit_estimators()
         self.estimators_scores = self.score_estimators()
+        print("F1-scores and their Jaccard similarity:")
+        print(self.scores_df)
```
Comment (Baukebrenninkmeijer) [inline table_evaluator/table_evaluator.py:500]: The scores_df should be the new estimator_scores imo. We should remove the old estimator_scores and use this as the new one. Furthermore, we should integrate the score from these scores into the similarity score, similar to what the old estimator_scores was doing. 

Can you propose a way to integrate this nicely? Does using the 1-MAPE/MSE still make sense for these cases?
